File: 0398-random-pick-index/0398-random-pick-index.java
class Solution {
    Map<Integer, List<Integer>> map;

    public Solution(int[] nums) {
        this.map = new HashMap();
        for(int i=0; i<nums.length; i++) {
            if(!map.containsKey(nums[i])) {
                map.put(nums[i], new ArrayList());
            }
            map.get(nums[i]).add(i);
        }
    }
    
    public int pick(int target) {
        List<Integer> arr = map.get(target);
        int rand = (int) Math.floor(Math.random() * arr.size());
        return arr.get(rand);
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int param_1 = obj.pick(target);
 */

File: 0398-random-pick-index/0398-random-pick-index.ts
class Solution {
    nums = null
    constructor(nums: number[]) {
        this.nums = nums
    }

    pick(target: number): number {
        let res = 0
        let count = 0
        for(let i=0; i<this.nums.length; i++) {
            if(this.nums[i] === target) {
                count++
                const rand = Math.ceil(Math.random() * count)
                if(rand === count) {
                    res = i
                }
            }
        }
        return res
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * var obj = new Solution(nums)
 * var param_1 = obj.pick(target)
 */